packages/common/src/converter/search-attribute-payload-converter.ts

```diff
@@ -0,0 +1,34 @@
+import { ValueError } from '@temporalio/internal-workflow-common';
+import { PayloadConverter } from './payload-converter';
+import { JsonPayloadConverter } from './json-payload-converter';
+import { Payload, str } from './types';
+
+const jsonConverter = new JsonPayloadConverter();
+
+/**
+ *
+ */
+export class SearchAttributePayloadConverter implements PayloadConverter {
+  public toPayload(value: unknown): Payload | undefined {
```

Comment:
We should verify only valid types are passed in here (only arrays with single type)